### PR TITLE
fix: update pre-selected target process

### LINKED ISSUE #19171: Instance migration is not triggered with pre-selected target process
### Describe the bug

If a target process version is pre-selected in instance migration mode, the migration operation is not triggered after confirming the migration.

This bug was introduced with the recently added migration target pre-selection feature (https://github.com/camunda/camunda/issues/18761). 

### To Reproduce

1. Go to instances list view
2. Select a process which has multiple versions from the filters panel (e.g. complex process in demo data)
3. Select the first version of that process from the filters panel
4. Select one or more process instances from the list
5. Click Migrate button
6. Click Next button (without changing anything else)
7. Click Confirm button

### Current behavior

- The migration operation is not triggered (there is no migration entry in the operations panel).

### Expected behavior

- The migration operation is triggered and the user can observe the status in the operations panel.

### Workaround

This bug happens only if the target process is pre-selected and is not changed. As soon as the user changes the target process or version manually, the instance migration will be triggered after clicking "Confirm" button.

As a workaround the user can change the target version (or process) to a different one and than back to the desired one.

### Environment

- Operate Version: 8.6.0-alpha2

### Additional context

---------------------------------------------------------------------------------------------

### Definition of Ready - Checklist

- [ ] The bug has been reproduced by the assignee in the environment compatible with the provided one; otherwise, the issue is closed with a comment
- [ ] The issue has a meaningful title, description, and testable acceptance criteria
- [ ] The issue has been labeled with an appropriate `Bug-area` label
- [ ] Necessary screenshots, screen recordings, or files are attached to the bug report

For UI changes required to solve the bug:

- [ ] Design input has been collected by the assignee

#### :mag: Root Cause Analysis

The reason for this is that the target process definition key is set in the onChange function of the process/version dropdown fields. If the process version is pre-selected, the onChange function is never called.

### :point_right: Handover Dev to QA

- Changed components:
- Side effects on other components:
- Handy resources:
  BPMN/DMN models, plugins, scripts, REST API endpoints + example payload, etc :
  
- Example projects:

- Commands/Steps needed to test; Versions to validate:

- Docker file / HELM chart : in case that it needed to be tested via docker share the version contain the fixed along with version of other services .

- Release version ( in which version this fixed/feature will be released):

### PR DESCRIPTION
**Pull request by pedesen**
## Description

- Move setTargetProcessDefinitionKey call from onChange to useEffect to ensure it is called every time selectedTargetProcessId changes
- Refactor e2e test setup
- Add e2e test for auto-mapping

### Code review hints 

The flow is now the following:

#### Manual target selection
- The user selects a process and version using the combobox/dropdown :arrow_right: process and version are set in `processes.migration` store
- This updates the `selectedTargetProcessId` computed value in `processes.migration` store
- This triggers the useEffect in `TargetDiagram` which calls `processInstanceMigrationStore.setTargetProcessDefinitionKey`

#### Pre-selection
- The migration mode is entered
- This triggers version selection autorun in `processes.migration` store
- This calls setSelectedTargetProcess and setSelectedTargetVersion actions in `processes.migration` store
- This updates the `selectedTargetProcessId` computed value in `processes.migration` store
- This triggers the useEffect in `TargetDiagram` which calls `processInstanceMigrationStore.setTargetProcessDefinitionKey`

## Related issues

closes #19171

### CODE REVIEW

File: operate/client/src/App/Processes/MigrationView/BottomPanel/index.test.tsx [hunk 1]
```diff
@@ -55,11 +55,14 @@ const Wrapper = ({children}: Props) => {
...
-    processXmlMigrationTargetStore.reset();
-    processStatisticsStore.reset();
-  });
+    return () => {
+      processInstanceMigrationStore.reset();
+      processXmlMigrationSourceStore.reset();
+      processXmlMigrationTargetStore.reset();
+      processStatisticsStore.reset();
+    };
+  }, []);
```
Comment (pedesen) [inline operate/client/src/App/Processes/MigrationView/BottomPanel/index.test.tsx:64]: This is unrelated. I just noticed it was not right. Not it is ensured that the stores are reset on unmount only.

File: operate/client/src/App/Processes/MigrationView/Footer/index.test.tsx [hunk 2]
```diff
@@ -21,7 +21,7 @@ const Wrapper = ({children}: Props) => {
   useEffect(() => {
     processInstanceMigrationStore.setCurrentStep('elementMapping');
     return processInstanceMigrationStore.reset;
-  });
+  }, []);
```
Comment (pedesen) [inline operate/client/src/App/Processes/MigrationView/Footer/index.test.tsx:24]: also unrelated. Now it is ensured that the effect runs only once (at mount)

File: operate/client/src/App/Processes/MigrationView/TopPanel/Diagrams/TargetDiagram.tsx [hunk 3]
```diff
@@ -41,6 +41,9 @@ const TargetDiagram: React.FC = observer(() => {
     } else {
       processXmlStore.reset();
     }
+    processInstanceMigrationStore.setTargetProcessDefinitionKey(
+      selectedTargetProcessId ?? null,
+    );
```
Comment (pedesen) [inline operate/client/src/App/Processes/MigrationView/TopPanel/Diagrams/TargetDiagram.tsx:46]: This is the actual fix. This was moved from `TargetVersionField.tsx`

File: operate/client/src/App/Processes/MigrationView/TopPanel/Diagrams/TargetVersionField.tsx [hunk 4]
```diff
@@ -32,9 +32,6 @@ const TargetVersionField: React.FC = observer(() => {
           if (!isNil(selectedItem)) {
             processInstanceMigrationStore.resetFlowNodeMapping();
             processesStore.setSelectedTargetVersion(selectedItem);
-            processInstanceMigrationStore.setTargetProcessDefinitionKey(
-              processesStore.selectedTargetProcessId ?? null,
-            );
```
Comment (pedesen) [inline operate/client/src/App/Processes/MigrationView/TopPanel/Diagrams/TargetVersionField.tsx:37]: This was moved to `TargetDiagram.tsx`

**Review by vitorwtavares — APPROVED** [review]

**Comment by pedesen** [comment]
@Szik Please test locally using this docker image name: `registry.camunda.cloud/team-operate/camunda-operate:branch-fe-19171-fix-pre-selected-process`

**Review by Szik — APPROVED** [review]
